File: 0071-simplify-path/0071-simplify-path.cpp
class Solution {
public:
    string simplifyPath(string path) {
        vector<string> v;
        string ans="",temp="";
        for(auto x: path){
            if(x=='/'){
                if(temp==".."){
                    if(v.size()==0)temp="";
                    else v.pop_back();
                    temp="";
                }
                else if(temp==".") temp="";
                if(temp=="")continue;
                v.push_back(temp);
                temp="";
            }
            else temp +=x;
        }
        if(temp.length()!=0 && temp!="."){
            if(temp==".."){
                    if(v.size()==0)temp="";
                    else v.pop_back();
                }else
            v.push_back(temp);
        }
        for(auto x:v)ans +='/'+x;
        return (ans.length()==0) ? "/":ans;

    }
};

File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        path = [p for p in path.split('/') if p]
        stack = []
        for p in path:
            if p=='.':continue
            if p == '..':
                if stack:stack.pop()
            else:
                stack.append(p)
        return '/'+'/'.join(stack)
        